dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/utils/DataRequestProcessingUtils.kt

```diff
@@ -249,6 +249,46 @@ class DataRequestProcessingUtils
             }
         }
 
+        /**
+         * Checks whether a request already exists on Dataland in a non-final status (i.e. in status "Open" or "Answered")
+         * and returns the request id
+         * @param companyId the company ID of the data request
+         * @param framework the framework of the data request
+         * @param reportingPeriod the reporting period of the data request
+         * @return the requestId if a request in non-final status exists, else null
+         */
+        fun getRequestIdForDataRequestWithNonFinalStatus(
+            companyId: String,
+            framework: DataTypeEnum,
+            reportingPeriod: String,
+        ): String? {
+            val openDataRequests =
+                findAlreadyExistingDataRequestForCurrentUser(
+                    companyId, framework, reportingPeriod, RequestStatus.Open,
+                )
+
+            val answeredDataRequests =
+                findAlreadyExistingDataRequestForCurrentUser(
+                    companyId, framework, reportingPeriod, RequestStatus.Answered,
+                )
+
+            openDataRequests?.takeIf { it.size > 1 }?.let {
+                throw ConflictApiException(
+                    "More than one open data request.",
```

Comment:
Now that the bug is fixed I am almost certain it will trigger this check. We need to handle this differently.